components/ntp_background_images/browser/ntp_sponsored_images_data.cc

```diff
@@ -145,25 +148,70 @@ NTPSponsoredImagesData::NTPSponsoredImagesData(
     return;
   }
 
+  url_prefix = base::StringPrintf("%s://%s/", content::kChromeUIScheme,
+                                  kBrandedWallpaperHost);
   if (auto* name = json_value->FindStringKey(kThemeNameKey)) {
     theme_name = *name;
-    DVLOG(2) << __func__ << ": Theme name: " << theme_name;
+    url_prefix += kSuperReferralPath;
+  } else {
+    url_prefix += kSponsoredImagesPath;
+  }
+
+  auto* campaigns_value = json_value->FindListKey(kCampaignsKey);
+  if (campaigns_value) {
+    ParseCampaignsList(campaigns_value->Clone(), installed_dir);
+  } else {
+    // Get a global campaign directly if the campaign list doesn't exist.
+    const auto campaign =
+        GetCampaignFromValue(json_value->Clone(), installed_dir);
+    if (campaign.IsValid())
+      campaigns.push_back(campaign);
+  }
```

Comment:
nit: how about a `VLOG` with the result of the parsing? As this is from external data, might help debug issues. Or do we have that somewhere else?

Reply:
Added log by https://github.com/brave/brave-core/pull/11628